Sources/PubNubChatComponents/Message List/MessageListViewController.swift

```diff
@@ -133,4 +209,162 @@ open class ChatViewController: CollectionViewComponent {
+      view.backgroundColor = .clear
+      view.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(onTapGestureRecognized(_:))))
+      
+      return view
+    }()
+    
+    init(
+      presentedViewController: UIViewController,
+      presentingViewController: UIViewController?,
+      messageCell: MessageListItemCell,
+      messageReactionsComponentSize: CGSize
+    ) {
+      self.messageCell = messageCell
+      self.messageReactionsComponentSize = messageReactionsComponentSize
+      
+      super.init(
+        presentedViewController: presentedViewController,
+        presenting: presentingViewController
+      )
+    }
+    
+    override var frameOfPresentedViewInContainerView: CGRect {
+      
+      guard let containerView = containerView, let messageCell = messageCell else {
+        return super.frameOfPresentedViewInContainerView
+      }
+      
+      let messageCellFrame = messageCell.superview?.convert(messageCell.frame, to: containerView) ?? .zero
+      
+      // Presents the reaction components beyond the cell that was pressed.
+      // If it doesn't fit at the screen, then the picker component is presented at the bottom of a view in which a presentation occurs
+      let x = messageCellFrame.minX
+      let y = min(containerView.frame.maxY - messageReactionsComponentSize.height - containerView.safeAreaInsets.bottom, messageCellFrame.maxY + 10)
+      
+      return CGRect(
+        x: x,
+        y: y,
+        width: messageReactionsComponentSize.width,
+        height: messageReactionsComponentSize.height
+      )
+    }
+    
+    @objc
+    func onTapGestureRecognized(_ sender: UIGestureRecognizer) {
+      presentedViewController.dismiss(animated: true)
+    }
+    
+    override func presentationTransitionWillBegin() {
+      guard let containerView = containerView else {
+        return
+      }
+      containerView.addSubview(dimmingView)
+      containerView.sendSubviewToBack(dimmingView)
+    }
+    
+    override func dismissalTransitionDidEnd(_ completed: Bool) {
+      if completed {
+        dimmingView.removeFromSuperview()
+      }
+    }
+  }
+  
+  // MARK: UIViewControllerAnimatedTransitioning aka animation controller
+  
+  @objc
+  class AnimationController : NSObject, UIViewControllerAnimatedTransitioning {
+    
+    private let duration: TimeInterval
+    private let isBeingPresented: Bool
+    
+    init(duration: TimeInterval, isBeingPresented: Bool) {
+      self.duration = duration
+      self.isBeingPresented = isBeingPresented
+    }
+    
+    func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
+      return duration
+    }
+    
+    func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+      
+      let currentViewController = transitionContext.viewController(forKey: .from)!
+      let finalViewController = transitionContext.viewController(forKey: .to)!
```

Comment:
I would check if it is nil, I know it is not likely to occur but potential crash scares me so much

Reply:
Ok, I added a `guard` statement to check this